### ISSUE TITLE
SQLiteDiskIOException: disk I/O error (code 1802) during Account.getStats()

### ISSUE THREAD

**Issue by rogerval** [k9mail/k-9#3058]
Please search to check for an existing issue (including closed issues, for which the fix may not have yet been released) before opening a new issue: https://github.com/k9mail/k-9/issues?q=is%3Aissue

### Expected behavior
Tell us what should happen
K9 should start and display list of accounts

### Actual behavior
Tell us what happens instead
Home screen goes blank for a second (icons disappear), then home screen returns to normal. K9 doesn't appear to start.
### Steps to reproduce
1. Attempt to start k9
2. Wait a second, while screen goes blank, then returns to normal.

### Environment
K-9 Mail version: 5.403

Android version: 7 (Moto G5)

Account type (IMAP, POP3, WebDAV/Exchange): 6 POP3 accounts.

(I won't mind at all if this issue is moved, merged, etc. I wasn't sure where to post it, as it's a beta version I believe.)
See attached log file (luckily I hadn't turned logging off before I upgraded to the beta) :
[Text of bug report for k-9 email 180107.txt](https://github.com/k9mail/k-9/files/1609181/Text.of.bug.report.for.k-9.email.180107.txt)

Hope this helps,
Roger V.

**Comment by rogerval** [comment]
My bad... K9 didn't crash after giving phone a reboot.

**Comment by rogerval** [comment]
Sorry to say, version 5.403 does crash after all, but not always immediately.
Mostly I can read emails, download new emails, and generally look around the accounts, and k9 works properly.
Sometimes when it crashes it occurs after I've opened k9.
Other times, it crashes without appearing to open at all (see my first entry for this issue) and this appears in the log file: "android.database.sqlite.SQLiteException: not an error (code 0): Could not open the database in read/write mode."
Once this happens, I can't open QuickPic (a photo viewer) without getting a "toast" with the same message on it. (I don't think QuickPic has a bug; just mentioning it here as a symptom.)
Force stopping both k9 and QuickPic doesn't fix it, and I have to re-start the phone.
See attached log file.
[Text of bug report for k-9 email 180108.txt](https://github.com/k9mail/k-9/files/1610175/Text.of.bug.report.for.k-9.email.180108.txt)

Thanks in advance,
Roger V.

**Comment by philipwhiuk** [comment]
```
01-08 05:56:26.533 16612 16612 E AndroidRuntime: Process: com.facebook.mlite, PID: 16612
01-08 05:56:26.533 16612 16612 E AndroidRuntime: java.lang.RuntimeException: Unable to create application com.facebook.mlite.MLiteApplication: java.lang.RuntimeException: java.io.FileNotFoundException: /mnt/expand/5519beb1-c901-4ec2-8afe-a0fbd2207b46/user/0/com.facebook.mlite/lib-main/dso_lock (Read-only file system)
```
Interesting error there from Facebook Lite. `Read-only file system` never a great sign.

I wonder if we've accidentally introduced multi-threaded DB access @cketti ?

https://stackoverflow.com/questions/21554277/android-sqlite-database-sqlitediskioexception-disk-i-o-error-code-1802

**Comment by cketti** [comment]
That sounds more like a problem with your device rather than a bug in K-9 Mail. Did you extend your internal storage with an SD card? If so, maybe the SD card is faulty?

**Comment by rogerval** [comment]
@cketti Thanks for your observation. :-) 
Yes, I did extend the storage with an SD card in late August last year. It's a Kingston, so hopefully hasn't failed yet. (And the other k9 crashes I've had problems with started in July last year, before I put the card in.) The Android OS did complain that the card was slow when I installed it but other than that, it went in without issue. (It's supposed to be class 10 which I thought was rather good, but I'm no SD card expert... )

@philipwhiuk Thanks to you, too. :-)
You've made a very interesting comment about Messenger Lite; it has crashed a couple of times, but I didn't connect that with a possible card fault.

I'll see if I can find an SD card test utility on Google Play (perhaps an exercise in futility... :-) )
